src/Network/NWFramer.cs

```diff
@@ -399,23 +399,24 @@ public bool ParseOutput (nuint minimumIncompleteLength, nuint maximumLength, Mem
 		[DllImport (Constants.NetworkLibrary)]
 		static extern unsafe bool nw_framer_parse_input (OS_nw_framer framer, nuint minimum_incomplete_length, nuint maximum_length, byte *temp_buffer, ref BlockLiteral parse);
 
-		delegate void nw_framer_parse_input_t (IntPtr block, IntPtr buffer, nuint buffer_length, bool is_complete);
+		delegate int nw_framer_parse_input_t (IntPtr block, IntPtr buffer, nuint buffer_length, bool is_complete);
 		static nw_framer_parse_input_t static_ParseInputHandler = TrampolineParseInputHandler;
 
 		[MonoPInvokeCallback (typeof (nw_framer_parse_input_t))]
-		static void TrampolineParseInputHandler (IntPtr block, IntPtr buffer, nuint buffer_length, bool is_complete)
+		static int TrampolineParseInputHandler (IntPtr block, IntPtr buffer, nuint buffer_length, bool is_complete)
 		{
-			var del = BlockLiteral.GetTarget<Action<Memory<byte>, bool>> (block);
+			var del = BlockLiteral.GetTarget<Func<Memory<byte>, bool, int>> (block);
 			if (del != null) {
 				var bBuffer = new byte[buffer_length];
 				Marshal.Copy (buffer, bBuffer, 0, (int)buffer_length);
 				var mValue = new Memory<byte>(bBuffer);
-				del (mValue, is_complete);
+				return del (mValue, is_complete);
 			}
+			return 0;
 		}
 
 		[BindingImpl (BindingImplOptions.Optimizable)]
-		public bool ParseInput (nuint minimumIncompleteLength, nuint maximumLength, Memory<byte> tempBuffer, Action<Memory<byte>, bool> handler)
+		public bool ParseInput (nuint minimumIncompleteLength, nuint maximumLength, Memory<byte> tempBuffer, Func<Memory<byte>, bool, int> handler)
```

Comment:
`Func<Memory<byte>, bool, int>` -> use a custom delegate because I can't guess that the arguments should be